src/lib.rs

```diff
@@ -299,92 +322,206 @@ pub fn write_b(cw: &mut ContextWriter, fi: &FrameInvariants, fs: &mut FrameState
+    if bsize >= BlockSize::BLOCK_8X8 &&
+       (bsize == BlockSize::BLOCK_8X8 || partition != PartitionType::PARTITION_SPLIT) {
+        cw.bc.update_partition_context(&bo, subsize, bsize);
+    }
+}
+
+fn write_sb_no_split_64x64(fi: &FrameInvariants, fs: &mut FrameState, cw: &mut ContextWriter,
+                            bsize: BlockSize, bo: &BlockOffset) {
+    assert!(bo.x + MAX_MIB_SIZE <= cw.bc.cols);
+    assert!(bo.y + MAX_MIB_SIZE <= cw.bc.rows);
+
+    let partition = PartitionType::PARTITION_NONE;
+    cw.write_partition(bo, partition, bsize);
+
+    let rdo_none = rdo_mode_decision(fi, fs, cw, bsize, bo);
+
+    write_b(fi, fs, cw, rdo_none.pred_mode, bsize, bo);
+
+    let subsize = get_subsize(bsize, partition);
+
+    if bsize >= BlockSize::BLOCK_8X8 &&
+       (bsize == BlockSize::BLOCK_8X8 || partition != PartitionType::PARTITION_SPLIT) {
+        cw.bc.update_partition_context(&bo, subsize, bsize);
+    }
 }
 
 fn encode_tile(fi: &FrameInvariants, fs: &mut FrameState) -> Vec<u8> {
     let w = ec::Writer::new();
     let fc = CDFContext::new();
-    let bc = BlockContext::new(fi.sb_width*16, fi.sb_height*16);
+    let bc = BlockContext::new(fi.w_in_b, fi.h_in_b);
     let mut cw = ContextWriter {
         w: w,
         fc: fc,
         bc: bc,
     };
 
-    let q = dc_q(fi.qindex) as f64;
-    let q0 = q / 8.0_f64;	// Convert q into Q0 precision, given thatn libaom quantizers are Q3.
-
-    // Lambda formula from doc/theoretical_results.lyx in the daala repo
-    let lambda = q0*q0*2.0_f64.log2()/6.0;	// Use Q0 quantizer since lambda will be applied to Q0 pixel domain
-
     for sby in 0..fi.sb_height {
         cw.bc.reset_left_contexts();
 
         for sbx in 0..fi.sb_width {
             let sbo = SuperBlockOffset { x: sbx, y: sby };
-            let po = sbo.plane_offset(&fs.input.planes[0].cfg);
-            let tell = cw.w.tell_frac();
-
-            let mut best_mode = PredictionMode::DC_PRED;
-            let mut best_rd = std::f64::MAX;
-
-            for &mode in RAV1E_INTRA_MODES {
-                let checkpoint = cw.checkpoint();
+            let bo = sbo.block_offset(0, 0);
 
-                write_sb(&mut cw, fi, fs, &sbo, mode, BlockSize::BLOCK_64X64);
-                let d = sse_64x64(&fs.input.planes[0].slice(&po), &fs.rec.planes[0].slice(&po));
-                let r = ((cw.w.tell_frac() - tell) as f64)/8.0;
+            // TODO(anyone):
+            // RDO-based block size decision for each SuperBlock can be done here.
 
-                let rd = (d as f64) + lambda*r;
-                if rd < best_rd {
-                    best_rd = rd;
-                    best_mode = mode;
-                }
-
-                cw.rollback(checkpoint.clone());
+            // Encode SuperBlock
+            //if sby < fi.sb_height-1 && sbx < fi.sb_width-1 {
+            if false {
+                write_sb_no_split_64x64(fi, fs, &mut cw, BlockSize::BLOCK_64X64, &bo);
```

Comment:
I think we can just delete the old write_sb_no_split_64x64 code.

Reply:
Sure- I kept it there since enabling it does not work (open bug), and fixing it now,
But, I'd better not distract reviewers, so removing it.